apps/ewallet/lib/ewallet/fetchers/amount_fetcher.ex

```diff
@@ -15,86 +19,82 @@ defmodule EWallet.AmountFetcher do
      "'amount' not allowed when exchanging values. Use from_amount and/or to_amount."}
   end
 
-  def fetch(%{"amount" => amount}, from, to) when is_integer(amount) do
-    {:ok, Map.put(from, :from_amount, amount), Map.put(to, :to_amount, amount), %{}}
-  end
-
   def fetch(%{"amount" => amount}, from, to) when is_binary(amount) do
     handle_string_amount(amount, fn amount ->
       {:ok, Map.put(from, :from_amount, amount), Map.put(to, :to_amount, amount), %{}}
     end)
   end
 
-  def fetch(%{"amount" => amount}, _from, _to) do
-    {:error, :invalid_parameter, "'amount' is not a number: #{amount}"}
+  def fetch(%{"amount" => amount}, from, to) when is_integer(amount) do
+    {:ok, Map.put(from, :from_amount, amount), Map.put(to, :to_amount, amount), %{}}
   end
 
-  def fetch(%{"from_amount" => from_amount, "to_amount" => to_amount}, from, to)
-      when is_number(from_amount) and is_number(to_amount) do
-    {:ok, Map.put(from, :from_amount, from_amount), Map.put(to, :to_amount, to_amount), %{}}
+  def fetch(%{"amount" => amount}, _from, _to) do
```

Comment:
Is binary amount handled in functions above this? (like in the calculation controller)